src/rotater.c

```diff
@@ -0,0 +1,25 @@
+#include <malloc.h>
+#include "rotater.h"
+
+int oLookUp(int** o, int searchChar, int i) {
+    if(!searchChar || !i) return 0;
+    return o[searchChar-1][i-1];
+}
+
+int jump(int bwtIndex, int jumpChar, int* c, int** o) {
+    return c[jumpChar] + oLookUp(o, jumpChar, bwtIndex);
+}
+
+struct Range* rotateString(const int* string, int stringLen, int* c, int** o, int btwLen) {
+    struct Range* r = malloc(sizeof(*r));
```

Comment:
I think I would prefer to get the `Range *` pointer as an argument. Here, you end up allocating a shitload of `struct Range` during a typical run, which is potentially very time consuming (and if you forget to free them, you have a real problem). If you add the range object as an argument, the caller can reuse one allocated `struct Range` multiple times, and you can even put it on the stack at the caller so you avoid `malloc()` allocations for this object entirely.